Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-204.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: 4f46693867b001869e522683a46a71902b4d7639
Containerized:False
Number of Error(s): 6
src-react/components/Monitoring.tsx(3,10): error TS2724: \@hpcc-js/comms\ has no exported member named \MachineServiceEx\. Did you mean \MachineService\?
src/DiskUsage.ts(79,119): error TS2554: Expected 1 arguments, but got 2.
src/DiskUsage.ts(276,52): error TS2559: Type string[] has no properties in common with type GetTargetClusterUsageRequest.
make[2]: *** [esp/src/CMakeFiles/eclwatch.dir/build.make:330: esp/src/build/dist/stub.eclwatch.js] Error 1
make[1]: *** [CMakeFiles/Makefile2:12228: esp/src/CMakeFiles/eclwatch.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2022-03-10_19-57-47
time logs started
Prep time   :    17 sec (00:00:17)
Build time  :   166 sec (00:02:46)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   183 sec (00:03:03)
time logs end

[Build error:](https://gist.github.com/edb989c00c024329b7ec7580f807f240)
Build log [Build log](https://gist.github.com/HPCCSmoketest/edb989c00c024329b7ec7580f807f240#file-result-22-03-10-19-54-27-log)

--- a/esp/src/src-react/components/Menu.tsx
+++ b/esp/src/src-react/components/Menu.tsx
@@ -138,7 +138,7 @@ const subMenuItems: SubMenuItems = {
         { headerText: nlsHPCC.ClusterProcesses + " (L)", itemKey: "/processes" },
         { headerText: nlsHPCC.SystemServers + " (L)", itemKey: "/servers" },
         { headerText: nlsHPCC.Security + " (L)", itemKey: "/security" },
-        { headerText: nlsHPCC.Monitoring + " (L)", itemKey: "/monitoring" },
+        { headerText: nlsHPCC.Monitoring, itemKey: "/monitoring" },
         { headerText: nlsHPCC.DESDL + " (L)", itemKey: "/desdl" },
     ],
 };

--- a/esp/src/src-react/components/Monitoring.tsx
+++ b/esp/src/src-react/components/Monitoring.tsx
@@ -1,0 +1,143 @@
+import * as React from "react";
+import { CommandBar, ContextualMenuItemType, ICommandBarItemProps, Image, Link } from "@fluentui/react";
+import { MachineService } from "@hpcc-js/comms";
+import { ShortVerticalDivider } from "./Common";
+import { tree } from "./DojoGrid";
+import { useFluentGrid } from "../hooks/grid";
+import { HolyGrail } from "../layouts/HolyGrail";
+import * as Utility from "src/Utility";
+import nlsHPCC from "src/nlsHPCC";
+
+function getStatusImageName(row) {
+    switch (row.Status) {
+        case "Error":
+            return "error.png";
+        case "Warning":
+            return "warning.png";
+        case "Normal":
+        default:
+            return "normal.png";
+    }
+}
+
+const machine = new MachineService({ baseUrl: "" });
+
+const defaultUIState = {
+    hasSelection: false
+};
+
+interface MonitoringProps {
+
+}
+
+export const Monitoring: React.FunctionComponent<MonitoringProps> = ({
+
+}) => {
+
+    const [uiState, setUIState] = React.useState({ ...defaultUIState });
+    const [data, setData] = React.useState<any[]>([]);
+
+    //  Grid ---
+    const { Grid, selection, copyButtons } = useFluentGrid({
+        data,
+        primaryID: "__hpcc_id",
+        sort: { attribute: "Name", "descending": false },
+        filename: "monitoring",
+        columns: {
+            StatusID: { label: "", width: 0, sortable: false, hidden: true },
+            ComponentType: tree({
+                label: "Name", sortable: true, width: 200,
+                formatter: function (Name, row) {
+                    return <Image src={Utility.getImageURL(getStatusImageName(row))} />;
+                }
+            }),
+            StatusDetails: { label: "Details", sortable: false },
+            URL: {
+                label: "URL", width: 200, sortable: false,
+                formatter: function (Name, row) {
+                    if (Name) {
+                        return <Link href={`http://${Name}`} target="_blank">{Name}</Link>;
+                    } else {
+                        return "";
+                    }
+                }
+            },
+            EndPoint: { label: "IP", sortable: true, width: 140 },
+            TimeReportedStr: { label: "Time Reported", width: 140, sortable: true },
+            Status: {
+                label: nlsHPCC.Severity, width: 130, sortable: false,
+                formatter: function (object, value, node, options) {
+                    switch (value) {
+                        case "Error":
+                            node.classList.add("ErrorCell");
+                            break;
+                        case "Warning":
+                            node.classList.add("WarningCell");
+                            break;
+                        case "Normal":
+                            node.classList.add("NormalCell");
+                            break;
+                    }
+                    node.innerText = value;
+                }
+            }
+        }
+    });
+
+    const refreshData = React.useCallback(() => {
+        machine.GetComponentStatus({
+            request: {}
+        }).then(({ ComponentStatusList }) => {
+            const componentStatuses = ComponentStatusList?.ComponentStatus;
+            if (componentStatuses) {
+                setData(componentStatuses.map((status, idx) => {
+                    return status?.StatusReports?.StatusReport.map((statusReport, idx) => {
+                        return {
+                            __hpcc_parentName: status.ComponentType + status.EndPoint,
+                            __hpcc_id: status.ComponentType + status.EndPoint + "_" + idx,
+                            ComponentType: statusReport.Reporter,
+                            Status: statusReport.Status,
+                            StatusDetails: statusReport.StatusDetails,
+                            URL: statusReport.URL
+                        };
+                    });
+                }).flat());
+            }
+        });
+    }, []);
+
+    //  Command Bar  ---
+    const buttons = React.useMemo((): ICommandBarItemProps[] => [
+        {
+            key: "refresh", text: nlsHPCC.Refresh, iconProps: { iconName: "Refresh" },
+            onClick: () => refreshData()
+        },
+        { key: "divider_1", itemType: ContextualMenuItemType.Divider, onRender: () => <ShortVerticalDivider /> },
+        {
+            key: "open", text: nlsHPCC.Open, disabled: !uiState.hasSelection, iconProps: { iconName: "WindowEdit" },
+            onClick: () => {
+                if (selection.length === 1) {
+                    window.location.href = `http://${selection[0].Name}`;
+                } else {
+                    for (let i = 0; i < selection.length; ++i) {
+                        window.open(`http://${selection[i].Name}`, "_blank");
+                    }
+                }
+            }
+        }
+    ], [refreshData, selection, uiState.hasSelection]);
+
+    //  Selection  ---
+    React.useEffect(() => {
+        const state = { ...defaultUIState };
+        state.hasSelection = selection.length > 0;
+        setUIState(state);
+    }, [selection]);
+
+    return <HolyGrail
+        header={<CommandBar items={buttons} farItems={copyButtons} />}
+        main={
+            <Grid />
+        }
+    />;
+};

--- a/esp/src/src-react/routes.tsx
+++ b/esp/src/src-react/routes.tsx
@@ -193,7 +193,7 @@ export const routes: RoutesEx = [
     },
     {
         mainNav: ["topology"],
-        path: "/monitoring", action: () => import("./layouts/DojoAdapter").then(_ => <_.DojoAdapter widgetClassID="MonitoringWidget" />)
+        path: "/monitoring", action: () => import("./components/Monitoring").then(_ => <_.Monitoring />)
     },
     {
         mainNav: ["topology"],

